modules/dind/examples_test.go

```diff
@@ -0,0 +1,51 @@
+package dind_test
+
+import (
+	"context"
+	"fmt"
+	"github.com/docker/docker/client"
+	"github.com/testcontainers/testcontainers-go/modules/dind"
+	"log"
+
+	"github.com/testcontainers/testcontainers-go"
+)
+
+func ExampleRun() {
+	// runDinDContainer {
+	ctx := context.Background()
+
+	dindContainer, err := dind.Run(ctx, "docker:28.0.1-dind")
+	defer func() {
+		if err := testcontainers.TerminateContainer(dindContainer); err != nil {
+			log.Printf("failed to terminate container: %s", err)
+		}
+	}()
+	if err != nil {
+		log.Printf("failed to start container: %s", err)
+		return
+	}
+
+	// Retrieve the host where the DinD daemon is listening
+	host, err := dindContainer.Host(ctx)
+	if err != nil {
+		log.Printf("failed to get docker host: %s", err)
+		return
+	}
+
+	cli, err := client.NewClientWithOpts(client.WithHost(host), client.WithAPIVersionNegotiation())
```

Comment:
suggestion: wdyt if the module provides a method to get a docker client with that host? I'd use the `testcontainers.NewDockerClientWithOpts` function so that the client is using the same client infra.

Thoughts?

Reply:
I thought of that, but then we're adding an external dependency to the docker cli with a specific version.

We can do that and make sure the CLI version stars the same with the DinD image version.

Reply:
More than that, users that will use the DinD module, will probably already have a the docker client dependency matching the version their project use.

If we provide a docker client, it will be with the version of the library and could prevent some issues to be detected.

I like that the DinD module does not provide a docker client

Reply:
Makes sense, thanks for the explanations. It's exactly the same use case where we avoid adding client libraries in other modules, to allow consumers to declare their own ones.